client/client.go

```diff
@@ -0,0 +1,123 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/go-http-utils/headers"
+	"go.opencensus.io/plugin/ochttp"
+
+	"github.com/SKF/go-rest-utility/client/auth"
+)
+
+const (
+	DefaultUserAgent      string = "go-rest-utility/v1"
+	DefaultAcceptEncoding string = "gzip"
+)
+
+type Doer interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+type Client struct {
+	BaseURL       *url.URL
+	TokenProvider auth.TokenProvider
+
+	client Doer
```

Comment:
Is there any practical advantage of having an interface here instead of just *http.Client?

Reply:
If you want to use another Library as the inner client. This also opens up for mocking if we want to test the internals.

The only thing this library need is a Do function which takes a `http.Request` and somehow produce a `http.Response`.